exasol/ai/mcp/server/mcp_server.py

```diff
@@ -0,0 +1,271 @@
+            self.tool(
+                self.list_schemas,
+                description=(
+                    "Lists schemas in the Exasol Database. "
+                    "For each schema provides the name and an optional comment."
+                ),
+            )
+        if self.config.tables.enable or self.config.views.enable:
+            self.tool(
+                self.list_tables,
+                description=(
+                    "Lists tables in the specified schema of the Exasol Database. "
+                    "For each table provides the name and an optional comment."
+                ),
+            )
+        if self.config.functions.enable:
+            self.tool(
+                self.list_functions,
+                description=(
+                    "Lists functions in the specified schema of the Exasol Database. "
+                    "For each function provides the name and an optional comment."
+                ),
+            )
+        if self.config.scripts.enable:
+            self.tool(
+                self.list_scripts,
+                description=(
+                    "Lists the user defined functions (UDF) in the specified schema of "
+                    "the Exasol Database. For each function provides the name and an "
+                    "optional comment."
+                ),
+            )
+        if self.config.columns.enable:
+            self.tool(
+                self.describe_table,
+                description=(
+                    "Describes the specified table in the specified schema of the "
+                    "Exasol Database. Returns the list of table columns. For each "
+                    "column provides the name, the data type, an optional comment, "
+                    "the primary key flag and the foreign key flag."
+                ),
+            )
+
+    def _build_meta_query(
+        self, meta_name: str, conf: MetaSettings, schema_name: str, *predicates
+    ) -> str:
+        """
+        Builds a metadata query.
+
+        Args:
+            meta_name:
+                Must be one of "SCHEMA", "TABLE", "VIEW", "FUNCTION", or "SCRIPT".
+            conf:
+                Metadata type settings, which is a part of the server configuration.
+            schema_name:
+                The schema name provided in the call to the tool. Ignored if meta_name
+                =='SCHEMA'. Otherwise, if it's empty the current schema of the pyexasol
+                connection may be used. If there is no current schema, the query will
+                return objects from all schemas.
+            predicates:
+                Any additional predicates to be used in the WHERE clause.
+        """
+        predicates = [conf.select_predicate, *predicates]
+        if meta_name != "SCHEMA":
+            schema_name = schema_name or self.connection.current_schema()
+            if schema_name:
+                predicates.append(f"{meta_name}_SCHEMA = '{schema_name}'")
```

Comment:
If schema_name could be a user provided value, then we need to escape it

Reply:
All names then.